odoo/addons/base/models/avatar_mixin.py

```diff
@@ -0,0 +1,97 @@
+
+
+class AvatarMixin(models.AbstractModel):
+    _name = 'avatar.mixin'
+    _inherit = ['image.mixin']
+    _description = "Avatar Mixin"
+
+    # all image fields are base64 encoded and PIL-supported
+
+    avatar_1920 = fields.Image("Avatar", max_width=1920, max_height=1920, compute="_compute_avatar_1920", readonly=False)
+
+    avatar_1024 = fields.Image("Avatar 1024", max_width=1024, max_height=1024, compute="_compute_avatar_1024", readonly=False)
+    avatar_512 = fields.Image("Avatar 512", max_width=512, max_height=512, compute="_compute_avatar_512", readonly=False)
+    avatar_256 = fields.Image("Avatar 256", max_width=256, max_height=256, compute="_compute_avatar_256", readonly=False)
+    avatar_128 = fields.Image("Avatar 128", max_width=128, max_height=128, compute="_compute_avatar_128", readonly=False)
+
+    def _compute_avatar(self, avatar_field, image_field):
+        for record in self:
+            record[avatar_field] = record[image_field] or record._get_one_avatar(avatar_field, image_field)
+
+    def _get_one_avatar(self, avatar_field, image_field):
+        self.ensure_one()
+        if self[self._rec_name]:
+            return self._generate_avatar_svg(self[self._rec_name])
+        return self._get_avatar_placeholder()
+
+    @api.depends(lambda self: [self._rec_name, 'image_1920'])
+    def _compute_avatar_1920(self):
+        self._compute_avatar('avatar_1920', 'image_1920')
+
+    @api.depends(lambda self: [self._rec_name, 'image_1024'])
+    def _compute_avatar_1024(self):
+        self._compute_avatar('avatar_1024', 'image_1024')
+
+    @api.depends(lambda self: [self._rec_name, 'image_512'])
+    def _compute_avatar_512(self):
+        self._compute_avatar('avatar_512', 'image_512')
+
+    @api.depends(lambda self: [self._rec_name, 'image_256'])
+    def _compute_avatar_256(self):
+        self._compute_avatar('avatar_256', 'image_256')
+
+    @api.depends(lambda self: [self._rec_name, 'image_128'])
+    def _compute_avatar_128(self):
+        self._compute_avatar('avatar_128', 'image_128')
+
+    def _generate_avatar_svg(self, name):
+        if not name:
+            return self._get_avatar_placeholder()
+        base = sha512(name.encode('utf-8'))
+        initials = html_escape(self._generate_avatar_initials(name))
+        bgcolor = html_escape(self._generate_avatar_bgcolor(base))
+        return b64encode(bytes((
+            "<?xml version='1.0' encoding='UTF-8' ?>"
+            "<svg height='180' width='180' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink'>"
+            f"<rect fill='{bgcolor}' height='180' width='180'/>"
+            f"<text fill='#ffffff' font-size='96' text-anchor='middle' x='90' y='125' font-family='sans-serif'>{initials}</text>"
+            "</svg>"
+        ), 'utf-8'))
+
+    def _get_avatar_placeholder(self):
+        image_path = get_resource_path('base', 'static/img', 'avatar_grey.png')
+        return b64encode(file_open(image_path, 'rb').read())
```

Comment:
From what I remember this doesn't work because `base` in not in the normal `addons` module.
@leonardopavanrocha could you check again with this syntax to confirm?

Reply:
Sure it works, it's the [expected way to do it](https://github.com/odoo/odoo/blob/master/odoo/tools/misc.py#L179-L194) ;-)

```py
In [1]: from odoo import tools

In [2]: tools.file_open("base/static/img/avatar_grey.png", 'rb').read()[:8]
Out[2]: b'\x89PNG\r\n\x1a\n'
```